### PR TITLE
feat(S1): Allow filtering of channel pairs

### PR DESCRIPTION
**Pull request by carlosgjs**
Main change is to `src/noisepy/seis/S1_fft_cc_MPI.py` + a couple of small fixes

### CODE REVIEW

File: src/noisepy/seis/S1_fft_cc_MPI.py [hunk 1]
```diff
@@ -60,6 +61,12 @@ def cross_correlate(
         raw_store: Store to load data from
         fft_params: Parameters for the FFT calculations
         cc_store: Store for saving cross correlations
+        scheduler: Scheduler to use for parallelization
+        pair_filter: Function to decide whether a pair of channels should be used or not. E.g.
```
Comment (carlosgjs) [inline src/noisepy/seis/S1_fft_cc_MPI.py:65]: @mdenolle @niyiyu @kuanfufeng , I wanted to include a simple example. Does this one seem appropriate to you?

Comment (niyiyu) [reply src/noisepy/seis/S1_fft_cc_MPI.py:65]: Looks good to me!

Comment (mdenolle) [reply src/noisepy/seis/S1_fft_cc_MPI.py:65]: this looks like the good example we talked about yesterday!

**Review by mdenolle — APPROVED** [review]
looks good! i need to figure out what the context/spawn addition makes to the parallelization

**Comment by carlosgjs** [comment]
> 

@mdenolle https://britishgeologicalsurvey.github.io/science/python-forking-vs-spawn/

**Pull request merged**